### ISSUE TITLE
Q3 Roadmap 

### ISSUE THREAD

**Issue by dsikka** [vllm-project/llm-compressor#2919]
## FP4 + Rotations vLLM Support, Validation (QutLass), and Benchmarking

Support for QutLass already exists in vLLM but needs to be hooked up to the compressed-tensors integration for FP4 models. Once integrated, validation and benchmarking is required to compare its performance to the hadacore kernels.

### Status: In progress

Relevant PRs:
1. https://github.com/vllm-project/vllm/pull/43229 (includes current benchmarking results)
2. https://github.com/vllm-project/vllm/pull/43229
3. https://github.com/neuralmagic/vllm/pull/167

### Status: Not Yet Started

Currently, if a model does not have a definition in transformers, the only algorithms that are supported in LLM Compressor out-of-the-box are data-free pathways using the `model_free_ptq`.  Algorithms requiring data through oneshot (such as GPTQ or any activation-quantization scheme requiring calibration data) require custom transformers definition handling, as was done to support  [DeepSeek v3.2](https://github.com/vllm-project/llm-compressor/tree/main/src/llmcompressor/modeling/deepseekv32) for [NVFP4 quantization](https://github.com/vllm-project/llm-compressor/blob/main/examples/disk_offloading/deepseek_v32_example.py).

Options to extend support for models without native transformers definitions include:
- [ ] 1. A custom Pytorch definition and handling (i.e not using or requiring a transformers definition)
- [ ] 2. Directly introduce an observer pathway in vLLM that can capture statistics given a dataset

End outcome should ease the burden in quantizing certain models, such as Mistral or DeepSeek.

### Status: In Progress

With all the support landed to support NVFP4 generally across LLM Compressor as well as support for FP4 in vLLM, ablations are required to establish the preferred pathway and optimal recipes for NVFP4 quantization. This includes benchmarking MoE and non-MoE models across the following modifiers, using varying dataset types, observers (e.g iMatrix), etc. Note: the standard evals used should also include MRCR (or some other metric to capture long context performance) as well as new evals not captured (e.g. GPQA Diamond, BFCLv4)

Experiment and Metric Tracking: https://github.com/vllm-project/llm-compressor/issues/2921

### Small (non-MoE) Models (No Rotations)
Tracking: https://github.com/vllm-project/llm-compressor/issues/2922
Base Model: [RedHatAI/Llama-3.1-8B-Instruct](https://huggingface.co/RedHatAI/Llama-3.1-8B-Instruct)
- [x] RTN 
- [x] GPTQ
- [x] AWQ 
- [x] AutoRound 
- [x] Smoothquant

### MoE models (No Rotations) 
Tracking: https://github.com/vllm-project/llm-compressor/issues/3001
Base Models: Qwen3 MoE, Gemma4
- [x] RTN 
- [x] GPTQ
- [x] AWQ 
- [x] SmoothQuant
- [ ] AutoRound 

### Small (non-MoE) Models (Rotations)
- [ ] RTN 
- [ ] GPTQ
- [ ] AWQ 
- [ ] AutoRound 

### Status: In Progress

HIGGS is a mixed-precision quantization algorithm for large language models that uses individual layer MSE values alongside an ILP to identify the optimal per-layer quantization config for a given weight size budget.

#### Note
- #### We should extend the FP4 Baseline Benchmarking results with a mixed-precision recipe, on the RedHatAI/Llama-3.1-8B-Instruct model
- #### We should also validate end-to-end Humming algorithmic performance, using HIGGS. Are there other specific mixed-precision use-cases that we're interested in? (With AutoRound? With GPTQ? May just be whatever comes out of the FP4 baseline metrics)

Relevant PRs and RFCs:
1. https://github.com/vllm-project/llm-compressor/issues/2690

## Research Initiatives 
1. QAD Training Support 
   - [ ] Look into layer-wise QAD (fits into LLM-C and should give us better results compared to baseline PTQ; similar to AutoRound)
   - [ ] GSQ - Layerwise and then finetune for 2, 3 bits
  
Later (lower priority)  - WUSH Transforms: https://arxiv.org/abs/2512.00956

## AMD Assessment 
- [ ] Validate LLM Compressor and compressed-tensors support on AMD GPUs. Note: this will require looking into getting access to AMD as well.
- [ ] Assess support for compressed-tensors pathways on AMD in vLLM

### Kimi K3
1. https://huggingface.co/RedHatAI/Kimi-K3-NVFP4
2. https://huggingface.co/RedHatAI/Kimi-K3-FP8-BLOCK

### GLM 5.2
1. https://huggingface.co/RedHatAI/GLM-5.2-NVFP4-FP8
2. https://huggingface.co/RedHatAI/GLM-5.2-NVFP4

### Muse Glimmer
1. https://huggingface.co/RedHatAI/Muse-Glimmer-30B-NVFP4
2. https://huggingface.co/RedHatAI/Muse-Glimmer-30B-INT4
3. https://huggingface.co/RedHatAI/Muse-Glimmer-30B-FP8-block

### Nemotron 3 Lightning 
1. https://huggingface.co/RedHatAI/NVIDIA-Nemotron-3.5-Lightning-30B-A3B-FP8

### Qwen3.8 27b
1. https://huggingface.co/RedHatAI/Qwen3.8-27B-INT4
2. https://huggingface.co/RedHatAI/Qwen3.8-27B-NVFP4

### Qwen3.8 
1. https://huggingface.co/RedHatAI/Qwen3.8-2.4T-A95B-NVFP4-FP8
2. https://huggingface.co/RedHatAI/Qwen3.8-2.4T-A95B-NVFP4
